### PR TITLE
feat: change the operation name with more than one response body type

### PR DESCRIPTION
**Pull request by tadelesh**
Fix: https://github.com/Azure/autorest.go/issues/858

### CODE REVIEW

**Comment by chlowell** [comment]
Is it safe to rely on the order of body types in the swagger? For an operation consuming only JSON, we would generate a `Foo()` method the JSON body type. With this code, if a future version of the swagger adds `"consumes": ["image/jpeg", "application/json"]`, we would generate `Foo()` for image/jpeg and `FooWithJSON()`.

Similarly, isn't this a non-breaking change for the swagger?
```diff
---"consumes": ["application/json", "image/jpeg"]
+++"consumes": ["image/jpeg", "application/json"]
```

**Comment by tadelesh** [comment]
> Is it safe to rely on the order of body types in the swagger? For an operation consuming only JSON, we would generate a `Foo()` method the JSON body type. With this code, if a future version of the swagger adds `"consumes": ["image/jpeg", "application/json"]`, we would generate `Foo()` for image/jpeg and `FooWithJSON()`.
> 
> Similarly, isn't this a non-breaking change for the swagger?
> 
> ```diff
> ---"consumes": ["application/json", "image/jpeg"]
> +++"consumes": ["image/jpeg", "application/json"]
> ```

Because we already have shipped some packages with two consumes, I could only rely on the order to prevent breaking change for released packages. Regarding we must select one for non-suffix name operation, we could not relay on any sort from codegen side, so just leave it to the swagger author. Any sequence change will be detected by SDK breaking change tools, but yes, it is not a swagger breaking change.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Relying on the ordering is fragile so we shouldn't be going down that path.  We can discuss possible solutions in the linked issue.

File: test/autorest/optionalgroup/zz_generated_explicit_client.go [hunk 1]
```diff
@@ -341,28 +341,28 @@ func (client *ExplicitClient) postOptionalStringHeaderCreateRequest(ctx context.
 	return req, nil
 }
 
-// PostOptionalStringParameter - Test explicitly optional string. Please put null.
+// PostOptionalStringParameterWithText - Test explicitly optional string. Please put null.
```
Comment (jhendrixMSFT) [inline test/autorest/optionalgroup/zz_generated_explicit_client.go:344]: This feature should be off by default.  We shouldn't be changing any existing names.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

File: test/autorest/optionalgroup/zz_generated_explicit_client.go (hunk 1 above)
Comment (tadelesh) [reply test/autorest/optionalgroup/zz_generated_explicit_client.go:344]: Change default to `false`.

File: src/transform/transform.ts [hunk 2]
```diff
@@ -293,34 +293,60 @@ function recursiveAddMarshallingFormat(schema: Schema, marshallingFormat: 'json'
...
+            newOp.language.go!.name = name;
+            newOp.language.go!.protocolNaming = new protocolMethods(newOp.language.go!.name);
+            group.addOperation(newOp);
+            if (req.language.go!.description) {
+              req.language.go!.description = parseComments(req.language.go!.description);
+            }
+          }
+          group.operations.splice(group.operations.indexOf(op), 1);
+        }
+      } else {
```
Comment (jhendrixMSFT) [inline src/transform/transform.ts:340]: A lot of this code looks duplicated from the above branch.  Can this be consolidated?  The algorithm itself looks good.

Comment (tadelesh) [reply src/transform/transform.ts:315]: Resolved.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**